src/core/types/field-template.ts

```diff
@@ -0,0 +1,72 @@
+import { Nominal, range } from "../../utils";
+
+import { _FieldDecoderImpl } from "./field-decoder";
+
+//@ts-ignore
+export type _FieldTemplateImpl<T> = {
+  __path: string;
+};
+
+/**
+ * Pointer to a form field template.
+ * Used to interact with Formts validation API via <arrayField>.every() method.
+ */
+// @ts-ignore
+export interface FieldTemplate<T, Err = unknown>
+  extends Nominal<"FieldTemplate", Err> {}
+
+// prettier-ignore
+export type GenericFieldTemplate<T, Err = unknown> =
```

Comment:
the issue was with TS itself, `4.1.5` is not catching this bug, but bumping to `4.2.2` does
my VS code was always throwing errors since it's using TS version different from project